packages/bcer-data-portal/app/src/hooks/useLeaflet.tsx

```diff
@@ -0,0 +1,338 @@
+    turnCost: false,
+    localDistortionField: false,
+  };
+
+  const [routeOptions, setRouteOptions] = useState<RouteOptions>({
+    ...initialRoutingOptions,
+  });
+
+  /**
+   * Once Map is rendered in the dom this method must be called
+   * to set up map properties
+   */
+  const onRender = () => {
+    let m = L.map('map').setView([51.4, -116.9], 6);
+    // Saving map in state for future use
+    setMap(m);
+    addTileLayer(m);
+  };
+
+  /**
+   * Draws markers for all the locations
+   */
+  const drawAllMarkers = () => {
+    const mkrs: L.Marker[] = [];
+    mkrs.push(...drawStartingMarker());
+
+    if (selectedLocations?.length && map) {
+      selectedLocations.forEach((l) => {
+        if (l.latitude && l.longitude) {
+          let mkr = L.marker([+l.latitude, +l.longitude]);
+          mkr.bindTooltip(makeMarkerToolTip(l)).openTooltip();
+          mkrs.push(mkr);
+          mkr.addTo(map);
+        }
+      });
+    }
+    setMarkers(mkrs);
+    setMapBound(mkrs);
+  };
+
+  /**
+   * Appropriately zooms the map to show all markers
+   * @param mkrs
+   */
+  const setMapBound = (mkrs: L.Marker[]) => {
+    if (map && mkrs?.length) {
+      const group = new (L.featureGroup as any)(mkrs);
+      map.fitBounds(group.getBounds());
+    }
+  };
+
+  /**
+   * Draws a unique marker for the starting location
+   * @returns
+   */
+  const drawStartingMarker = (): L.Marker[] => {
+    if (startingLocation?.geometry?.coordinates && map) {
+      var startingIcon = new L.Icon({
+        iconUrl:
+          'https://raw.githubusercontent.com/pointhi/leaflet-color-markers/master/img/marker-icon-2x-red.png',
+        shadowUrl:
+          'https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png',
+        iconSize: [25, 41],
+        iconAnchor: [12, 41],
+        shadowSize: [41, 41],
+      });
+
+      const [lng, lat] = startingLocation.geometry.coordinates;
+
+      const startingMarker = L.marker([lat, lng], { icon: startingIcon });
+      startingMarker.bindTooltip(
+        `<div style="text-align:center;"><b>Starting Location</b><br/>${safeGuardHtmlString(
+          startingLocation.properties.fullAddress
+        )}</div>`
+      );
+      startingMarker.addTo(map);
+      return [startingMarker];
+    }
+    return [];
+  };
+
+  /**
+   * Makes toolTip content from retail location markers
+   * @param l
+   * @returns
+   */
+  const makeMarkerToolTip = (l: BusinessLocation): string => {
+    return `<div style="text-align:center;"><b>${safeGuardHtmlString(
+      l.business?.businessName ?? l.business?.legalName
+    )}</b><br/>${safeGuardHtmlString(l?.geoAddress ?? l.addressLine1)}</div>`;
+  };
+
+  /**
+   * Tool tip string is injected as html. This function makes sure to safe guard against user input.
+   * @param text
+   * @returns
+   */
+  const safeGuardHtmlString = (text: string) => {
+    return (
+      text?.replace(/[`~!@#$%^&*()_|+\=?;:'",.<>\{\}\[\]\\\/]/gi, '') ?? ''
```

Comment:
I would suggest that we use something like [sanitize-html](https://www.npmjs.com/package/sanitize-html) for this (without any accepted tags) to make sure that no edge cases are missed